### PR TITLE
feat: Always include failed account ID and region info along with failed resource ID

### LINKED ISSUE #584: Error Summary Doesn't Specify the Account
### Describe the bug

When errors occur we print out the details for the user. The details at this time do not include a region or an account so it makes it super difficult to pinpoint which account had the issue.

This wasn't as big of a deal when all accounts were manually specified. But as we are supporting `orgs` that have 100s of accounts it makes it impossible to locate... you end up having to enable debug logs for everything to find the source

```
Resource: apigateway.rest_apis Type: Resolving Severity: Error
        Summary: column resolver "model_template" failed for table "aws_apigateway_rest_api_models": operation error API Gateway: GetModelTemplate, https response error StatusCode: 400, RequestID: 1b0d6910-e3a4-4d3e-b61f-0b5552bcf21b, BadRequestException: Unsupported data type for mapping template: string
Resource: apigateway.rest_apis Type: Resolving Severity: Error
        Summary: column resolver "model_template" failed for table "aws_apigateway_rest_api_models": operation error API Gateway: GetModelTemplate, https response error StatusCode: 400, RequestID: 8e87a382-6ce1-4fea-acc1-3d2da05260bc, BadRequestException: Unsupported data type for mapping template: string
Resource: apigateway.rest_apis Type: Resolving Severity: Error
        Summary: column resolver "model_template" failed for table "aws_apigateway_rest_api_models": operation error API Gateway: GetModelTemplate, https response error StatusCode: 404, RequestID: 5bf1ea00-e5fa-4079-bfb7-9120645cd6c7, NotFoundException: Invalid model name specified: EnumS
```

### Expected Behavior

More information like 

### Steps to Reproduce

see output above

### Possible Solution

Include resource ARN (if available) or at least the info from the client (region, account)

### Provider and CloudQuery version

all

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by disq**
Better than https://github.com/cloudquery/cq-provider-sdk/pull/210

Fixes #584

### CODE REVIEW

**Review by bbernays — APPROVED** [review]

File: client/errors.go [hunk 1]
```diff
@@ -16,37 +16,76 @@ import (
...
+
+		// remove accountID and region from PK list as we always prepend them
+		for _, val := range d.Description().ResourceID {
+			if val != client.AccountID && val != client.Region {
+				resIdList = append(resIdList, val)
+			}
+		}
+
+		resIdOverride = diag.WithResourceId(resIdList)
+	}
```
Comment (roneli) [inline client/errors.go:35]: nit: maybe make this a function? and pass a WithAWSResoruceId(d) (do the logic you did here) and then you can call this function in each return making this function lighter

**Review by roneli — APPROVED** [review]
LGTM, had one nit.

File: client/errors.go (hunk 1 above)
Comment (disq) [reply client/errors.go]: fixed

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**